homeassistant/components/plugwise/config_flow.py

```diff
@@ -175,14 +176,15 @@ async def async_step_user(
 
             try:
                 api = await validate_gw_input(self.hass, user_input)
-            except InvalidSetupError:
-                errors[CONF_BASE] = "invalid_setup"
+            except ConnectionFailedError:
+                errors[CONF_BASE] = "cannot_connect"
             except InvalidAuthentication:
                 errors[CONF_BASE] = "invalid_auth"
-            except PlugwiseException:
-                errors[CONF_BASE] = "cannot_connect"
+            except (InvalidXMLError, ResponseError):
+                errors[CONF_BASE] = "response_error"
+            except UnsupportedDeviceError:
+                errors[CONF_BASE] = "warn_code_owner"
```

Comment:
A user should not warn the code-owner...

Reply:
See below for explanation.

If us codeowners don't know about the major firmware update, it would be nice to get a warning via a clear error. Like above.

Reply:
It's clear how/what to change.